cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1060,15 +1060,59 @@ pub(crate) fn emit(
                         prefix,
                         opcode,
                         2,
-                        reg_g.to_reg(),
+                        dst.to_reg(),
                         addr,
                         rex_flags,
                     );
                 }
             }
         }
 
-        Inst::XmmCmove { size, cc, src, dst } => {
+        Inst::CmoveOr {
+            size,
+            cc1,
+            cc2,
+            consequent,
+            alternative,
+            dst,
+        } => {
+            //let inst = inst.clone();
+            let first_cmove = Inst::Cmove {
+                cc: *cc1,
+                size: *size,
+                consequent: consequent.clone(),
+                alternative: alternative.clone(),
+                dst: dst.clone(),
+            };
+            first_cmove.emit(sink, info, state);
+
+            //let inst = inst.clone();
+            let second_cmove = Inst::Cmove {
+                cc: *cc2,
+                size: *size,
+                consequent: consequent.clone(),
+                alternative: alternative.clone(),
+                dst: dst.clone(),
```

Comment:
This overwrites `dst` from above, I think -- it seems we want the first cmove to write a tmp, and `alternative` in the second to be that tmp? So then we have the equivalent of `cmove(cc1, a, cmove(cc2, b, alternative))`.